### PR TITLE
[4.1] Expose group layouts for com_fields

### PR DESCRIPTION
**Pull request by dgrammatiko**
Pull Request is an alternative to https://github.com/joomla/joomla-cms/pull/36265.

### Summary of Changes

Add a layout option on the group level

### CODE REVIEW

File: administrator/components/com_fields/forms/group.xml [hunk 1]
```diff
@@ -154,6 +154,20 @@
 				<option value="0">JNO</option>
 				<option value="1">JYES</option>
 			</field>
+
+			<field
+				name="layout"
+				type="filelist"
+				label="COM_FIELDS_FIELD_LAYOUT_LABEL"
+				default="render"
+				directory="components/com_fields/layouts/fields"
```
Comment (laoneo) [inline administrator/components/com_fields/forms/group.xml:163]: What happens with the overrides in the component and templates? Should be covered also somehow.

Comment (brianteeman) [reply administrator/components/com_fields/forms/group.xml:163]: that's why I suggested a new fieldtype based on componentlayoutlist

Comment (dgrammatiko) [reply administrator/components/com_fields/forms/group.xml:163]: > that's why I suggested a new fieldtype based on componentlayoutlist

Unfortunately the `componentLayout` is only for `com_xxx/tmpl` if I read correctly the code

> What happens with the overrides in the component and templates? Should be covered also somehow.

 Overrides of these layouts are still possible on the template due to JLayouts
About `overrides in the component` I'm not sure what you mean but the code in the layouts it's literally the same so it shouldn't break anything. If the option `don't render anything` is on then all these are neglected, right?

**Comment by dgrammatiko** [comment]
I will let @ceford to take over here

**Comment by laoneo** [comment]
I would leave this one open here as alternative. The whole discussion in #36265 is about the old code, so we have a clear history.

**Comment by dgrammatiko** [comment]
ok, reopened

File: administrator/components/com_fields/forms/group.xml (hunk 1 above)
Comment (brianteeman) [reply administrator/components/com_fields/forms/group.xml:163]: > > that's why I suggested a new fieldtype based on componentlayoutlist

> Unfortunately the componentLayout is only for com_xxx/tmpl if I read correctly the code

ok "inspired by" then. we also have modulelayoutfield which is similar

File: components/com_fields/layouts/fields/DIV.php [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+{
+	$fields = $item->jcfields ?: FieldsHelper::getFields($context, $item, true);
+}
+
+if (empty($fields))
+{
+	return;
+}
+
+$output = array();
```
Comment (PhilETaylor) [inline components/com_fields/layouts/fields/DIV.php:52]:
```suggestion
$output = [];
```

File: components/com_fields/layouts/fields/DL.php [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+{
+	$fields = $item->jcfields ?: FieldsHelper::getFields($context, $item, true);
+}
+
+if (empty($fields))
+{
+	return;
+}
+
+$output = array();
```
Comment (PhilETaylor) [inline components/com_fields/layouts/fields/DL.php:52]:
```suggestion
$output = [];
```

File: components/com_fields/layouts/fields/DL.php [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+	{
+		continue;
+	}
+
+	$class = $field->name . ' ' . $field->params->get('render_class');
+	$layout = $field->params->get('layout', 'render');
+	$content = FieldsHelper::render($context, 'field.' . $layout, array('field' => $field));
+
+	// If the content is empty do nothing
+	if (trim($content) === '')
```
Comment (PhilETaylor) [inline components/com_fields/layouts/fields/DL.php:67]: https://3v4l.org/VDRBp

```suggestion
	if (!trim($content))
```

File: components/com_fields/layouts/fields/OL.php [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+{
+	$fields = $item->jcfields ?: FieldsHelper::getFields($context, $item, true);
+}
+
+if (empty($fields))
+{
+	return;
+}
+
+$output = array();
```
Comment (PhilETaylor) [inline components/com_fields/layouts/fields/OL.php:52]:
```suggestion
$output = [];
```

File: components/com_fields/layouts/fields/OL.php [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+	{
+		continue;
+	}
+
+	$class = $field->name . ' ' . $field->params->get('render_class');
+	$layout = $field->params->get('layout', 'render');
+	$content = FieldsHelper::render($context, 'field.' . $layout, array('field' => $field));
+
+	// If the content is empty do nothing
+	if (trim($content) === '')
```
Comment (PhilETaylor) [inline components/com_fields/layouts/fields/OL.php:67]:
```suggestion
	if (!trim($content))
```

**Comment by dgrammatiko** [comment]
Since @ceford already took over and this code was just to showcase the layout approach, this PR has served it's purpose

**Pull request closed without merging**